src/uu/factor/src/numeric.rs

```diff
@@ -172,55 +197,141 @@ impl Arithmetic for Montgomery {
     }
 }
 
+pub(crate) trait OverflowingAdd: Sized {
+    fn overflowing_add_(self, n: Self) -> (Self, bool);
+}
+impl OverflowingAdd for u32 {
+    fn overflowing_add_(self, n: Self) -> (Self, bool) {
+        self.overflowing_add(n)
+    }
+}
+impl OverflowingAdd for u64 {
+    fn overflowing_add_(self, n: Self) -> (Self, bool) {
+        self.overflowing_add(n)
+    }
+}
+impl OverflowingAdd for u128 {
+    fn overflowing_add_(self, n: Self) -> (Self, bool) {
+        self.overflowing_add(n)
+    }
+}
+
+pub(crate) trait Int:
+    AsPrimitive<u64>
+    + AsPrimitive<u128>
```

Comment:
I realize I’m the one who wrote this, but can you just remove `AsPrimitive` and use explicit methods instead for now (like `as_u64`/`to_u64`)?  I dislike how we use `as_` in only some places.

Doing so will also fix the type inference issue with `r_2`.

Reply:
A quick way to get this set up would be to split `Int` into:
```rust
// you should prob change the name btw (SingleInt is a little...)
pub(crate) trait Int: SingleInt {
    type Intermediate: SingleInt;

    fn as_intermediate(self) -> Self::Intermediate;
    fn from_intermediate(n: Self::Intermediate) -> Self;
}

pub(crate) trait SingleInt: Display + Debug + PrimInt + OverflowingAdd + WrappingNeg + WrappingSub + WrappingMul {
    fn as_u64(self) -> u64;
    fn as_u128(self) -> u128;
    fn as_u32(self) -> u32;
}

// impl SingleInt for u32, u64, u128
// impl Int for u32 and u64
```

... or something like that.  You can probably just macro away the implementations as they should be pretty simple.

Reply:
> I realize I’m the one who wrote this, but can you just remove `AsPrimitive` and use explicit methods instead for now (like `as_u64`/`to_u64`)?

Done

Reply:
> You can probably just macro away the implementations as they should be pretty simple.

I'm not sure it's worth doing: there are only 3 implementations of `Int` and 2 of `DoubleInt`; I might try in a bit, though, if only because it's an excuse to learn how to macro  :)

Reply:
That was actually much easier than I expected.

Reply:
Yeah, these sorts of macros are really easy to write, so they’re very convenient.